ocaml/typing/ctype.ml

```diff
@@ -3858,7 +3858,17 @@ let filter_arrow env t l ~force_tpoly =
     Tvar { layout } ->
       let t', arrow_desc = function_type (get_level t) in
       link_type t t';
-      constrain_type_layout_exn env Unify t' layout;
+
+      begin match constrain_type_layout env t' layout with
```

Comment:
I would prefer this before the call to `link_type`. It probably doesn't matter, but I would always want `link_type` to link two types with compatible layouts.

Reply:
I'd also drop the leading and trailing whitespace to match the style of the compiler.

Reply:
Changed the call order and removed the extra whitespaces.

There are two other cases of `link_type` being called before `constrain_type_layout_exn`:

https://github.com/ocaml-flambda/flambda-backend/blob/5f80067709edffb71ad1bd833dfd15c562c0e684/ocaml/typing/ctype.ml#L4374
https://github.com/ocaml-flambda/flambda-backend/blob/5f80067709edffb71ad1bd833dfd15c562c0e684/ocaml/typing/ctype.ml#L4390

Should we change those too?

Reply:
Yes, probably. Would be curious to know whether that changes error messages. My desire for the reordering is purely because it just makes more sense to me in the other order, but if it improves errors, that would be great!

Reply:
Ooh. Reordering made a nice improvement to error messages. Nice.